apache/trafficcontrol
# TP: adds first, inner and last header rewrite fields for topologies

## What does this PR (Pull Request) do?
This PR adds 3 new fields to the DNS* and HTTP* delivery service forms:

- firstHeaderRewrite
- innerHeaderRewrite
- lastHeaderRewrite

Validation was also added to:

a. only allow firstHeaderRewrite, innerHeaderRewrite and lastHeaderRewrite values if topology is selected for a ds
b. prevent edgeHeaderRewrite and midHeaderRewrite values if a topology is selected.

All 5 values are displayed to facilitate migration from edge/mid header rewrites to first/inner/last header rewrites when moving from explicit ds / server assignments towards using a topology instead.


## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
- Ensure that at least one topology exists at https://tp.domain.net/#!/topologies
- create/update a DNS* ds
- add values for edge/mid/first/inner/last header rewrites
- toggle between a topology and no topology assigned and notice the validation of those fields.
- ensure edge/mid header rewrites are not allowed when a topology is assigned
- ensure first/inner/last header rewrites are not allowed when NO topology is assigned
- do the same for an HTTP* ds

## The following criteria are ALL met by this PR

- [x] This PR does NOT include tests as these fields are optional and only required fields are currently under test.
- [x] This PR does NOT includes documentation changes as each ds form is not documented.
